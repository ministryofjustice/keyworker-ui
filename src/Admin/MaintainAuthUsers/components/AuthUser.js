import React from 'react'
import PropTypes from 'prop-types'
import GridRow from '@govuk-react/grid-row'
import GridCol from '@govuk-react/grid-col'
import Button from '@govuk-react/button'
import Table from '@govuk-react/table'
import { BLACK, GREY_3 } from 'govuk-colours'
import MessageBar from '../../../MessageBar'
import { authRoleListType, contextAuthUserType } from '../../../types'

const AuthUser = props => {
  const { roleList, handleRemove, handleAdd, contextUser } = props

  const results = roleList.map(a => (
    <Table.Row key={a.roleCode}>
      <Table.Cell>{a.roleName}</Table.Cell>
      <Table.Cell alignRight>
        <Button
          buttonColour={GREY_3}
          buttonTextColour={BLACK}
          mb={0}
          id={`remove-button-${a.roleCode}`}
          value={a.roleCode}
          onClick={handleRemove}
        >
          Remove
        </Button>
      </Table.Cell>
    </Table.Row>
  ))

  return (
    <div>
      <MessageBar {...props} />
      <GridRow mb={5}>
        <GridCol setWidth="two-thirds">
          <Table id="user-details">
            <Table.Row>
              <Table.CellHeader>Username</Table.CellHeader>
              <Table.Cell>{contextUser.username}</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.CellHeader>Name</Table.CellHeader>
              <Table.Cell>
                {contextUser.firstName} {contextUser.lastName}
              </Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.CellHeader>Email</Table.CellHeader>
              <Table.Cell>{contextUser.email}</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.CellHeader>Locked</Table.CellHeader>
              <Table.Cell>{contextUser.locked ? 'Yes' : 'No'}</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.CellHeader>Enabled</Table.CellHeader>
              <Table.Cell>{contextUser.enabled ? 'Yes' : 'No'}</Table.Cell>
            </Table.Row>
          </Table>
        </GridCol>
      </GridRow>

      <GridRow>
        <GridCol setWidth="one-half">
          <Table id="user-roles">
            <Table.Row>
              <Table.CellHeader>Current roles</Table.CellHeader>
              <Table.CellHeader>&nbsp;</Table.CellHeader>
            </Table.Row>
            {results.length > 0 ? (
              results
            ) : (
              <Table.Row>
                <Table.Cell>No roles found</Table.Cell>
                <Table.Cell>&nbsp;</Table.Cell>
              </Table.Row>
            )}
          </Table>
        </GridCol>
      </GridRow>

      <div>
        <Button id="add-button" onClick={handleAdd}>
          Add role
        </Button>
      </div>
    </div>
  )
}

AuthUser.propTypes = {
  contextUser: contextAuthUserType.isRequired,
  roleList: authRoleListType.isRequired,
  handleRemove: PropTypes.func.isRequired,
  handleAdd: PropTypes.func.isRequired,
}

export default AuthUser
