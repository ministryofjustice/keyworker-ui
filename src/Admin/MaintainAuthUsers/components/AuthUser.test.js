import React from 'react'
import renderer from 'react-test-renderer'
import Table from '@govuk-react/table'
import { MemoryRouter } from 'react-router-dom'
import AuthUser from './AuthUser'

const user = {
  username: 'joesmith',
  firstName: 'Joe',
  lastName: 'Smith',
  locked: false,
  enabled: true,
  email: 'joe.smith@justice.gov.uk',
  lastLoggedIn: '2019-02-02T12:05:10',
}
const groupManagerUser = {
  username: 'joesmith',
  firstName: 'Joe',
  lastName: 'Smith',
  locked: false,
  enabled: true,
  email: 'joe.smith@justice.gov.uk',
  maintainAuthUsers: false,
  lastLoggedIn: '2019-02-02T12:05:10',
}
const adminUser = {
  username: 'joesmith',
  firstName: 'Joe',
  lastName: 'Smith',
  locked: false,
  enabled: true,
  email: 'joe.smith@justice.gov.uk',
  maintainAuthUsers: true,
  lastLoggedIn: '2019-02-02T12:05:10',
}

describe('Auth user display', () => {
  const stubFunc = () => {}

  describe('no roles', () => {
    const wrapper = renderer.create(
      <MemoryRouter>
        <AuthUser
          user={adminUser}
          contextUser={user}
          roleList={[]}
          groupList={[]}
          handleRoleAdd={stubFunc}
          handleRoleRemove={stubFunc}
          handleGroupAdd={stubFunc}
          handleGroupRemove={stubFunc}
          handleEnable={stubFunc}
          handleDisable={stubFunc}
        />
      </MemoryRouter>
    )
    it('should render correctly with user and no roles', () => {
      expect(wrapper.toJSON()).toMatchSnapshot()
    })

    it('should display message about no roles found', () => {
      const tableRows = wrapper.root.findAllByType(Table.Row)
      // last table row should have message
      expect(tableRows[tableRows.length - 3].props.children[0].props.children).toEqual('No roles found')
    })
  })

  describe('no groups - add groups not available for Group Manager', () => {
    const wrapper = renderer.create(
      <MemoryRouter>
        <AuthUser
          user={adminUser}
          contextUser={user}
          roleList={[]}
          groupList={[]}
          handleRoleAdd={stubFunc}
          handleRoleRemove={stubFunc}
          handleEnable={stubFunc}
          handleDisable={stubFunc}
          handleGroupAdd={stubFunc}
          handleGroupRemove={stubFunc}
        />
      </MemoryRouter>
    )
    it('should render correctly with user and no groups', () => {
      expect(wrapper.toJSON()).toMatchSnapshot()
    })

    it('should display message about no groups found', () => {
      const tableRows = wrapper.root.findAllByType(Table.Row)
      // last table row should have message
      expect(tableRows[tableRows.length - 1].props.children[0].props.children).toEqual('No groups found')
    })
  })

  describe('no groups - add groups available for admin user', () => {
    const wrapper = renderer.create(
      <MemoryRouter>
        <AuthUser
          user={adminUser}
          contextUser={user}
          roleList={[]}
          groupList={[]}
          handleRoleAdd={stubFunc}
          handleRoleRemove={stubFunc}
          handleGroupAdd={stubFunc}
          handleGroupRemove={stubFunc}
          handleEnable={stubFunc}
          handleDisable={stubFunc}
        />
      </MemoryRouter>
    )
    it('should render correctly with user and no groups', () => {
      expect(wrapper.toJSON()).toMatchSnapshot()
    })

    it('should display message about no groups found', () => {
      const tableRows = wrapper.root.findAllByType(Table.Row)
      // last table row should have message
      expect(tableRows[tableRows.length - 1].props.children[0].props.children).toEqual('No groups found')
    })
  })

  it('should render correctly with user roles and groups - add groups not available for Group Manager', () => {
    const verifiedUser = { verified: true, ...user }
    const wrapper = renderer.create(
      <MemoryRouter>
        <AuthUser
          user={groupManagerUser}
          contextUser={verifiedUser}
          roleList={[{ roleCode: 'roleA', roleName: 'Role A' }, { roleCode: 'roleB', roleName: 'Role B' }]}
          groupList={[{ groupCode: 'group1', groupName: 'Group 1' }, { groupCode: 'group2', groupName: 'Group 2' }]}
          handleRoleAdd={stubFunc}
          handleRoleRemove={stubFunc}
          handleEnable={stubFunc}
          handleDisable={stubFunc}
          handleGroupAdd={stubFunc}
          handleGroupRemove={stubFunc}
        />
      </MemoryRouter>
    )

    expect(wrapper.toJSON()).toMatchSnapshot()
  })

  it('should render correctly with user roles and groups - add groups available for admin user', () => {
    const verifiedUser = { verified: true, ...user }
    const wrapper = renderer.create(
      <MemoryRouter>
        <AuthUser
          user={adminUser}
          contextUser={verifiedUser}
          roleList={[{ roleCode: 'roleA', roleName: 'Role A' }, { roleCode: 'roleB', roleName: 'Role B' }]}
          groupList={[{ groupCode: 'group1', groupName: 'Group 1' }, { groupCode: 'group2', groupName: 'Group 2' }]}
          handleRoleAdd={stubFunc}
          handleRoleRemove={stubFunc}
          handleGroupAdd={stubFunc}
          handleGroupRemove={stubFunc}
          handleEnable={stubFunc}
          handleDisable={stubFunc}
        />
      </MemoryRouter>
    )

    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
