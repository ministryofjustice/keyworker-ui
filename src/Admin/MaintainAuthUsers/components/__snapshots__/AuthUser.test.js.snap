// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth user display no roles should render correctly with user and no roles 1`] = `
<div>
  <div
    className="sc-bZQynM gYfOhL"
  >
    <div
      className="sc-gzVnrw hGCwOi"
    >
      <table
        className="sc-EHOje kCMYgv"
        id="user-details"
      >
        <tbody
          className="sc-bwzfXH uTxCW"
        >
          <tr
            className="sc-ifAKCX cFlEyZ"
          >
            <th
              className="sc-bxivhb kZbmwT"
            >
              Username
            </th>
            <td
              className="sc-bxivhb jMmQxS"
            >
              joesmith
            </td>
          </tr>
          <tr
            className="sc-ifAKCX cFlEyZ"
          >
            <th
              className="sc-bxivhb kZbmwT"
            >
              Name
            </th>
            <td
              className="sc-bxivhb jMmQxS"
            >
              Joe
               
              Smith
            </td>
          </tr>
          <tr
            className="sc-ifAKCX cFlEyZ"
          >
            <th
              className="sc-bxivhb kZbmwT"
            >
              Email
            </th>
            <td
              className="sc-bxivhb jMmQxS"
            >
              joe.smith@justice.gov.uk
            </td>
          </tr>
          <tr
            className="sc-ifAKCX cFlEyZ"
          >
            <th
              className="sc-bxivhb kZbmwT"
            >
              Locked
            </th>
            <td
              className="sc-bxivhb jMmQxS"
            >
              No
            </td>
          </tr>
          <tr
            className="sc-ifAKCX cFlEyZ"
          >
            <th
              className="sc-bxivhb kZbmwT"
            >
              Enabled
            </th>
            <td
              className="sc-bxivhb jMmQxS"
            >
              Yes
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    className="sc-bZQynM eXvviO"
  >
    <div
      className="sc-gzVnrw doLCDi"
    >
      <table
        className="sc-EHOje kCMYgv"
        id="user-roles"
      >
        <tbody
          className="sc-bwzfXH uTxCW"
        >
          <tr
            className="sc-ifAKCX cFlEyZ"
          >
            <th
              className="sc-bxivhb kZbmwT"
            >
              Current roles
            </th>
            <th
              className="sc-bxivhb kZbmwT"
            >
               
            </th>
          </tr>
          <tr
            className="sc-ifAKCX cFlEyZ"
          >
            <td
              className="sc-bxivhb jMmQxS"
            >
              No roles found
            </td>
            <td
              className="sc-bxivhb jMmQxS"
            >
               
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div>
    <button
      className="sc-htoDjs gxKpuV"
      disabled={false}
      id="add-button"
      onClick={[Function]}
    >
      Add role
    </button>
  </div>
</div>
`;

exports[`Auth user display should render correctly with user and roles 1`] = `
<div>
  <div
    className="sc-bZQynM gYfOhL"
  >
    <div
      className="sc-gzVnrw hGCwOi"
    >
      <table
        className="sc-EHOje kCMYgv"
        id="user-details"
      >
        <tbody
          className="sc-bwzfXH uTxCW"
        >
          <tr
            className="sc-ifAKCX cFlEyZ"
          >
            <th
              className="sc-bxivhb kZbmwT"
            >
              Username
            </th>
            <td
              className="sc-bxivhb jMmQxS"
            >
              joesmith
            </td>
          </tr>
          <tr
            className="sc-ifAKCX cFlEyZ"
          >
            <th
              className="sc-bxivhb kZbmwT"
            >
              Name
            </th>
            <td
              className="sc-bxivhb jMmQxS"
            >
              Joe
               
              Smith
            </td>
          </tr>
          <tr
            className="sc-ifAKCX cFlEyZ"
          >
            <th
              className="sc-bxivhb kZbmwT"
            >
              Email
            </th>
            <td
              className="sc-bxivhb jMmQxS"
            >
              joe.smith@justice.gov.uk
            </td>
          </tr>
          <tr
            className="sc-ifAKCX cFlEyZ"
          >
            <th
              className="sc-bxivhb kZbmwT"
            >
              Locked
            </th>
            <td
              className="sc-bxivhb jMmQxS"
            >
              No
            </td>
          </tr>
          <tr
            className="sc-ifAKCX cFlEyZ"
          >
            <th
              className="sc-bxivhb kZbmwT"
            >
              Enabled
            </th>
            <td
              className="sc-bxivhb jMmQxS"
            >
              Yes
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    className="sc-bZQynM eXvviO"
  >
    <div
      className="sc-gzVnrw doLCDi"
    >
      <table
        className="sc-EHOje kCMYgv"
        id="user-roles"
      >
        <tbody
          className="sc-bwzfXH uTxCW"
        >
          <tr
            className="sc-ifAKCX cFlEyZ"
          >
            <th
              className="sc-bxivhb kZbmwT"
            >
              Current roles
            </th>
            <th
              className="sc-bxivhb kZbmwT"
            >
               
            </th>
          </tr>
          <tr
            className="sc-ifAKCX cFlEyZ"
          >
            <td
              className="sc-bxivhb jMmQxS"
            >
              Role A
            </td>
            <td
              className="sc-bxivhb eFpOLU"
            >
              <button
                className="sc-htoDjs kiMCan"
                disabled={false}
                id="remove-button-roleA"
                onClick={[Function]}
                value="roleA"
              >
                Remove
              </button>
            </td>
          </tr>
          <tr
            className="sc-ifAKCX cFlEyZ"
          >
            <td
              className="sc-bxivhb jMmQxS"
            >
              Role B
            </td>
            <td
              className="sc-bxivhb eFpOLU"
            >
              <button
                className="sc-htoDjs kiMCan"
                disabled={false}
                id="remove-button-roleB"
                onClick={[Function]}
                value="roleB"
              >
                Remove
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div>
    <button
      className="sc-htoDjs gxKpuV"
      disabled={false}
      id="add-button"
      onClick={[Function]}
    >
      Add role
    </button>
  </div>
</div>
`;
